src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/TypeStyle/CSharpUseImplicitTypeHelper.cs

```diff
@@ -333,6 +331,19 @@ protected override bool AssignmentSupportsStylePreference(
             return declaredType != null && declaredType.Equals(initializerType);
         }
 
+        // This also applies to a lambda assigned to a variable.  This will have no conversion, but can be converted as
+        // long as the type is Func<> or Action<> as that's what the language will infer here.
+        if (!conversion.Exists && expression is LambdaExpressionSyntax && semanticModel.Compilation.LanguageVersion() >= LanguageVersion.CSharp10)
+        {
+            var initializerType = semanticModel.GetTypeInfo(expression, cancellationToken).Type;
+            return declaredType.Equals(initializerType) &&
+                declaredType is
+                {
+                    Name: nameof(Func<int>) or nameof(Action<int>),
```

Comment:
>int [](http://example.com/codeflow?start=38&length=3)

intentional <int>?

Reply:
Yes.  The language doesn't allow leaving out a type-parameter in nameof.  I championed and implemented the feature to allow eliding this.  But that comes in C# 14, which we are not on yet.